Renamed UI `Vector1` to `Float` / Renamed enum `Float` precision to `Single`  (V2)

sebastienlagarde opened the pull request:
Replace PR: https://github.com/Unity-Technologies/Graphics/pull/1004 with a more up to date version

### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

**This PR does mainly cosmetic changes for better undestanding for artists:**
* UI Only rename of `Vector 1` occurences to `Float`
* Renamed `Float` value of precision enums to `Single` (should do no halo as it's serialized as ints)

![image](https://user-images.githubusercontent.com/4037271/85541836-b75cbd00-b618-11ea-9ad5-10158c90f5f1.png)

![image](https://user-images.githubusercontent.com/4037271/85541810-add35500-b618-11ea-8dca-c42a0ad495c9.png)

- Alias to our node searcher so Vector1 will continue to bring up the new float node. 
- Add a note about the change to our 20.2 upgrade guide

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

sebastienlagarde commented:
So Yamato is green (all the failure are instability or already present on  master). The PR is good to go after @Unity-Technologies/gfx-qa-bellevue  and @sharlenet  review the doc.

@alindmanUnity have added the alias thing.

sharlenet commented on com.unity.render-pipelines.high-definition/Documentation~/Master-Node-Unlit.md:
```diff
@@ -27,10 +27,10 @@ The following table describes the input ports on an Unlit Master node, including
+| **AlphaClipThreshold**                    | Float    | Fragment  | Set the alpha value limit that HDRP uses to determine whether it should render each pixel of the Material. If the alpha value of the pixel is equal to or higher than this threshold then HDRP renders the pixel. If the value is lower than this threshold then HDRP does not render the pixel. This port only appears when you enable th ...
 | **Emission**                           	| Vector 3 | Fragment  | Defines the Material's emission color value. Expects positive values.|
-| **Shadow Tint**                              | Vector 4 | Fragment  | Defines the shadow color value with alpha value. Enable only if Shadow Matte enable cf. [Master node settings menu](#SettingsMenu)|
+| **Shadow Tint**                           | Vector 4 | Fragment  | Defines the shadow color value with alpha value. Enable only if Shadow Matte enable cf. [Master node settings menu](#SettingsMenu)|
```
I'm not quite sure what this means: "Enable only if Shadow Matte enable cf."

sebastienlagarde replied:
this is not related to the change in this PR but it mean that Shadow tint property is only having an effect if shadow matte property is enabled.

sharlenet replied:
I rephrased it as follows:
> Defines the shadow color value with alpha value. The **Shadow Tint** property only has an effect if you enable the **Shadow Matte** property located in the [Master node settings menu](#SettingsMenu).

sharlenet commented on com.unity.shadergraph/Documentation~/Vector-1-Node.md:
```diff
@@ -1,20 +1,20 @@
-# Vector 1 Node
```
I'd like to know whether we should rename this page to `Float.md` (as well as update all links to `Vector-1-Node.md`).

sebastienlagarde replied:
hmm... good point. We should rename it. Will you do it @sharlenet or you expect I do ?

sharlenet replied:
I've made the changes and will approve this PR.

LandonTownsendUnity reviewed (changes requested):
When you create a float it says "Vector1" instead of float
![image](https://user-images.githubusercontent.com/52939698/94293514-3b258e80-ff13-11ea-9ea0-44ea777a9719.png)

LandonTownsendUnity reviewed (approved):
![image](https://user-images.githubusercontent.com/52939698/94293681-82138400-ff13-11ea-8d14-2c8b335bd70c.png)
![image](https://user-images.githubusercontent.com/52939698/94293695-88096500-ff13-11ea-93e3-0dc5c6734109.png)
"Single" under precision color mode should appear blue, but instead has no associated color

LandonTownsendUnity reviewed (changes requested):
![image](https://user-images.githubusercontent.com/52939698/94293681-82138400-ff13-11ea-8d14-2c8b335bd70c.png)
![image](https://user-images.githubusercontent.com/52939698/94293695-88096500-ff13-11ea-93e3-0dc5c6734109.png)
"Single" under precision color mode should appear blue, but instead has no associated color

sebastienlagarde commented:
> When you create a float it says "Vector1" instead of float
> ![image](https://user-images.githubusercontent.com/52939698/94293514-3b258e80-ff13-11ea-9ea0-44ea777a9719.png)

good catch, I fixed it. Not sure to understand what you mean by the "blue" for single. Was it blue  on master?

sebastienlagarde commented:
> ![image](https://user-images.githubusercontent.com/52939698/94293681-82138400-ff13-11ea-8d14-2c8b335bd70c.png)
> ![image](https://user-images.githubusercontent.com/52939698/94293695-88096500-ff13-11ea-93e3-0dc5c6734109.png)
> "Single" under precision color mode should appear blue, but instead has no associated color

@LandonTownsendUnity  so I checked on master and I have this:

![image](https://user-images.githubusercontent.com/21238527/94300073-c67d3f00-ff68-11ea-85b1-ac6aa069c18a.png)


I don't see the difference

sebastienlagarde commented:
> > When you create a float it says "Vector1" instead of float
> > ![image](https://user-images.githubusercontent.com/52939698/94293514-3b258e80-ff13-11ea-9ea0-44ea777a9719.png)
> 
> good catch, I fixed it. Not sure to understand what you mean by the "blue" for single. Was it blue on master?

This is now fixed

cdxntchou reviewed (approved):
and search synonyms!

sebastienlagarde commented:
Merged master with Antoine PR's change and fixed conflict. All good now. Yamato is running

sebastienlagarde commented:
@LandonTownsendUnity ok, Antoine helped me to understand the issue with the blue line you where referening. All is fixed now. Reviewed, Yamatoed. So ready for your QA review. 
I have tested various cases. But I think we need to check migration with precision difference.

Thanks

![image](https://user-images.githubusercontent.com/21238527/94951803-377ea280-04e5-11eb-9a88-9e38a54434a7.png)

LandonTownsendUnity reviewed (approved):
Did some further testing, didn't find any other problems.
